src/votekit/ballot_generator.py

```diff
@@ -1201,4 +1201,144 @@ def __init__(self, **data):
         k = len(data["candidates"])
         # Call the parent class's __init__ method to handle common parameters
         super().__init__(k=k, **data)
-        
\ No newline at end of file
+        
+
+class SlatePreference(BallotGenerator):
+    """
+    Class for generating ballots using a slate preference model. 
+    This model first samples a ballot type by flipping a cohesion parameter weighted coin.
+    It then fills out the ballot type via sampling with out replacement from the interval.
+
+    Only works with 2 blocs at the moment.
+    
+    Can be initialized with an interval or can be
+    constructed with the Dirichlet distribution using the `from_params` method in the
+    `BallotGenerator` class.
+
+    **Attributes**
+
+    `slate_to_candidates`
+    :   dictionary mapping of slate to candidates (ex. {bloc: [candidate]}).
+
+    `pref_interval_by_bloc`
+    :   dictionary mapping of bloc to preference interval.
+        (ex. {bloc: {candidate : interval length}}).
+
+    `bloc_voter_prop`
+    :   dictionary mapping of bloc to voter proportions (ex. {bloc: proportion}).
+
+    `cohesion_parameters`
+    :   dictionary mapping of slate to cohesion level (ex. {bloc: .7}).
+
+    **Methods**
+
+    See `BallotGenerator` base class
+    """
+
+    def __init__(self, slate_to_candidates: dict = {}, cohesion_parameters: dict = {}, **data):
```

Comment:
these default parameters should not be {}, see https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python for a good explanation as to why.

This will lead to strange hard-to-diagnose bugs.  Instead default to `None`, and have a conditional inside the constructor that sets to `{}` if `None` was passed.

Reply:
Turns out these shouldn't be optional anyway, but noted for the future!